utils/file_utils.py

```
@@ -41,3 +42,19 @@ def read_s3_duckdb(path,file_format):
+            df.write_parquet(path,storage_options=storage_options,mode="overwrite")
+        elif file_format == "csv":
+            df.write_csv(path,storage_options=storage_options,mode="overwrite")
+        elif file_format == "json":
+            df.write_json(path,storage_options=storage_options,mode="overwrite")
+        elif file_format == "delta":
+            df.write_delta(path,storage_options=storage_options,mode="overwrite")
+        else:
+            raise Exception("Invalid file format")
+        return df
+    except Exception as e:
+        raise Exception(f"Error occurred while writing to path: {e}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)